testing/ef-tests/src/result.rs

```diff
@@ -90,7 +97,13 @@ pub(crate) fn categorize_results(
     for case in results {
         match case.result.as_ref().err() {
             Some(Error::Skipped) => skipped.push(case),
-            Some(_) => failed.push(case),
+            Some(_) => {
+                if case.should_fail {
+                    passed.push(case);
+                } else {
+                    failed.push(case);
+                }
+            }
```

Comment:
This is why I say its hacky/inconsistent -- tests that should be skipped return Error::Skipped whereas should_fail right now adds a new method to the Case trait.

Its possible to do the same as skipped, by catching the error in `Case::run` and then checking if it should fail and return Error::ShouldFail -- it starts to look a bit unwieldly mainly because of all of the nesting but that can be fixed by ceating functions for some of the code

Reply:
Tests are passing, so I will push a less hackier way